### PR TITLE
Batch construct batches

### PR DESCRIPTION
**Pull request by sipa**
Reuse the serialization buffers used for constructing the LevelDB write batch. This avoids 2 allocations per utxo write.

I have not benchmarked this exact change, but in another branch I'm working on it results in a 12% speedup for the flush.

### CODE REVIEW

**Comment by gmaxwell** [comment]
ACK.  (I tested this on the above mentioned other branch.)

**Comment by laanwj** [comment]
Seems straightforward - code review ACK https://github.com/bitcoin/bitcoin/pull/9346/commits/ed5896976a5cb6d893babedec2bc88cb7a251033

File: src/dbwrapper.h [hunk 1]
```diff
@@ -53,38 +53,41 @@ class CDBBatch
...
-        CDataStream ssKey(SER_DISK, CLIENT_VERSION);
         ssKey.reserve(DBWRAPPER_PREALLOC_KEY_SIZE);
         ssKey << key;
         leveldb::Slice slKey(&ssKey[0], ssKey.size());
 
-        CDataStream ssValue(SER_DISK, CLIENT_VERSION);
         ssValue.reserve(DBWRAPPER_PREALLOC_VALUE_SIZE);
         ssValue << value;
         ssValue.Xor(dbwrapper_private::GetObfuscateKey(parent));
         leveldb::Slice slValue(&ssValue[0], ssValue.size());
```
Comment (laanwj) [inline src/dbwrapper.h]: CDataStream should really get a `.data()`. Not necessarily here though :)

**Review by laanwj — APPROVED** [review]

**Comment by paveljanik** [comment]
ACK https://github.com/bitcoin/bitcoin/pull/9346/commits/ed5896976a5cb6d893babedec2bc88cb7a251033

File: src/dbwrapper.h (hunk 1 above)
Comment (sipa) [inline src/dbwrapper.h]: Agree, but let's do that separately (#9353?).

**Pull request merged**